modules/importer/src/runner/clearly_defined/walker.rs

```diff
@@ -0,0 +1,208 @@
+use crate::runner::common::Error;
+use crate::runner::progress::{Progress, ProgressInstance};
+use crate::runner::report::{Phase, ReportBuilder};
+use serde::{Deserialize, Serialize};
+use std::collections::{HashSet, LinkedList};
+use std::io::{BufRead, Read};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tokio_util::bytes::Buf;
+use trustify_entity::labels::Labels;
+use trustify_module_ingestor::service::{Format, IngestorService};
+
+pub struct ClearlyDefinedWalker<P: Progress + Send + 'static> {
+    continuation: ClearlyDefinedItemContinuation,
+    source: String,
+    ingestor: IngestorService,
+    progress: P,
+    progress_instance: Option<P::Instance>,
+    report: Arc<Mutex<ReportBuilder>>,
+    coordinates_seen_this_run: HashSet<String>,
+}
+
+impl<P: Progress + Send + 'static> ClearlyDefinedWalker<P> {
+    pub fn new(
+        source: impl Into<String>,
+        ingestor: IngestorService,
+        report: Arc<Mutex<ReportBuilder>>,
+        progress: P,
+    ) -> Self {
+        Self {
+            continuation: Default::default(),
+            source: source.into(),
+            ingestor,
+            progress,
+            progress_instance: None,
+            report,
+            coordinates_seen_this_run: Default::default(),
+        }
+    }
+
+    pub fn continuation(mut self, continuation: ClearlyDefinedItemContinuation) -> Self {
+        self.continuation = continuation;
+        self
+    }
+
+    pub async fn run(mut self) -> Result<ClearlyDefinedItemContinuation, Error> {
+        let changes = reqwest::get(self.changes_index_url()).await?;
```

Comment:
This creates a new connection pool for each request. I think we should create and keep a client.

Same for the other calls.